### PR TITLE
added support for sqlalchemy and google authentication

### PR DESCRIPTION
**Pull request by ndliberial**
# Requirements
When creating your Pull request, please fill out the template below:

# PR details
_Is this a fix/ update/ new feature?_
Fix
_Include a short description_
Updated db access to use sqlalchemy and added support for google authentication
_Include links to JIRA tickets_

# Testing
_Have you tested it?_
Yes
# Assign to the weekly GitHub reviewer
_If you are a member of Ensembl, please check the Genebuild weekly Rotas and assign this week's GitHub reviewer to the PR_
@thibauthourlier

### CODE REVIEW

**Review by thibauthourlier — CHANGES_REQUESTED** [review]
There are few debugging prints which I haven't flagged but can be removed. Logging with a proper `verbose` flag is totally fine
I'm not sure why the requirements.txt file is mentionned but it probably shouldn't be there as the file in feature/main_alpha has no empty lines
The script seems very similar to https://github.com/Ensembl/ensembl-genes/pull/21. If it does something different, fine, otherwise it feels that it should be in the ensembl-genes repo and be as a library or a script with multiple parameters
It would be nice to have some documentation for the methods/functions

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 1]
```diff
@@ -12,18 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import mysql.connector
-from mysql.connector import Error
+#import mysql.connector
+#from mysql.connector import Error
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:16]: Done

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 2]
```diff
@@ -12,18 +12,37 @@
...
 import datetime
 import argparse
+import sqlalchemy as db
+from sqlalchemy import exc
 #import traceback
+import pymysql
+pymysql.install_as_MySQLdb()
 
 def fetch_db_data(query,database,host,port,user,password):
+  registry_db_url = 'mysql://' + user + ':' + password + '@' + host + ':' + port + '/' + database
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:31]: Done

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 3]
```diff
@@ -12,18 +12,37 @@
...
 import gspread
 from oauth2client.service_account import ServiceAccountCredentials
 import pprint
 import unicodedata
 import time
 import datetime
 import argparse
+import sqlalchemy as db
+from sqlalchemy import exc
 #import traceback
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:26]: Done

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 4]
```diff
@@ -12,18 +12,37 @@
...
+import pymysql
+pymysql.install_as_MySQLdb()
 
 def fetch_db_data(query,database,host,port,user,password):
+  registry_db_url = 'mysql://' + user + ':' + password + '@' + host + ':' + port + '/' + database
+  print(registry_db_url)
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:32]: Done

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 5]
```diff
@@ -12,18 +12,37 @@
...
+  try:
+    engine = db.create_engine(registry_db_url)
+    with engine.connect() as connection:
+      rows = connection.execute(db.text(query))
+
+  except exc.SQLAlchemyError as e:
+        print(type(e))
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:39]: Changed to exit function and display the error

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 6]
```diff
@@ -12,18 +12,37 @@
...
+      rows = connection.execute(db.text(query))
+
+  except exc.SQLAlchemyError as e:
+        print(type(e))  
+
+  engine.dispose()
+
+  return rows
+
+def fetch_db_data1(query,database,host,port,user,password):
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:45]: Done

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 7]
```diff
@@ -51,25 +70,25 @@ def update_assembly_sheet(assembly_db_data,meta_db_data,existing_sheet_records,a
   # dicts key on the versioned GCA (which is unique). Once the dicts are generated keys in the
   # assembly db dict are compared to the keys in the sheets dict. If a key is not in the sheets
   # dict then it is a new entry and gets made into a new row and added to the sheet. If the key
-  # is present then some tests are done to see if anything needs updating. Some of these tests
+  # is present then some tests are done to see of anything needs updating. Some of these tests
   # could be made generic, but there are some complex cases like when the filters need updating
   # The filters are basically tags for the assemblies that are then used to create the filter
   # views in sheets
-  min_contig_n50_filter = 100000
+  min_contig_n50_filter = 30000
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:77]: The reason for the change is to allow us cater for more assemblies outside the 100mb threshold especially now that we are working on some non-verts as well.

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 8]
```diff
@@ -100,23 +120,34 @@ def update_assembly_sheet(assembly_db_data,meta_db_data,existing_sheet_records,a
...
-      gettime = time.time()
-      # Find a workbook by name and open the first sheet
-      # Make sure you use the right name here.
-      assembly_sheet = client.open(worksheet_name).worksheet("EnsemblAssemblyRegistry")
+    #Check that time since last authentication is < 1hr
+    if(time.time() - gettime > 60* 59):#If greater than 1 hr, then re-authenticate
+        print("Updating time: " + gca)
+         # use creds to create a client to interact with the Google Drive API
+        scope = ['https://spreadsheets.google.com/feeds',
+           'https://www.googleapis.com/auth/drive']
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:134]: Reformatted and parsed as a parameter in subsequent method

File: scripts/genebuild/update_assembly_registry_sheet.py [hunk 9]
```diff
@@ -100,23 +120,34 @@ def update_assembly_sheet(assembly_db_data,meta_db_data,existing_sheet_records,a
...
+        assembly_sheet = client.open(worksheet_name).sheet1
+        #response = client.login()
+        #if response:
+          #  print ("Success line ")
+          #  print (response)
+           # return response
+        #else:
+         #   print ("Error line ")
+          #  print(response)
+       # gettime = time.time()#Get time of re-authentication
```
[the remark this answers is not in the document]
Comment (ndliberial) [reply scripts/genebuild/update_assembly_registry_sheet.py:150]: Done

**Comment by thibauthourlier** [comment]
@williamstark01 git says that there are no new changes since my last review and it looks like git is correct

**Comment by williamstark01** [comment]
@thibauthourlier you are right, there are no commits, I saw the resolved comments and mistaken then for code updates in the PR.

@ndliberial are there any updates to push maybe?

**Comment by williamstark01** [comment]
@ndliberial are there updates for this PR?

**Pull request closed without merging**